app/src/main/java/com/projectit3048c/MainViewModel.kt

```diff
@@ -141,4 +145,25 @@ class MainViewModel(var foodService : IFoodService =  FoodService()) : ViewModel
             handle.addOnFailureListener { Log.e("Firebase", "Save failed $it ") }
         }
     }
+
+    fun fetchPhotos() {
+        photos.clear()
+        user?.let {
+            user ->
+            var photoCollection = firestore.collection("users").document(user.uid).collection("specimens").document(selectedFoodAmount.foodId).collection("photos")
+            var photosListener = photoCollection.addSnapshotListener {
+                querySnapshot, firebaseFirestoreException ->
+                querySnapshot?.let {
+                    querySnapshot ->
+                    var documents = querySnapshot.documents
+                    documents?.forEach {
+                        var photo = it.toObject(Photo::class.java)
+                        photo?.let {
+                            photos.add(photo)
```

Comment:
Didn't rename it variable, needs to be:
photo?.let {
    photo ->
    photo.add(photo)
}